src/main/java/com/ldtteam/structurize/management/StructureName.java

```diff
@@ -106,13 +108,18 @@ else if (name.startsWith(Structures.SCHEMATICS_CACHE + Structures.SCHEMATICS_SEP
         {
             hut = schematic.split("\\d+")[0];
             section = Structures.SCHEMATICS_PREFIX;
-            if (ForgeRegistries.BLOCKS.getValue(new ResourceLocation(Constants.MINECOLONIES_MOD_ID, "blockhut" + hut)) != Blocks.AIR || HUTS.contains(hut))
-            {
-                section = hut;
-            }
-            else
+            //This allows mods other than minecolonies to create hut buildings and use the buildtool with them.
+            for (String origin : StructureLoadingUtils.getOriginMods().keySet())
             {
-                hut = "";
+                if (IAnchorBlock.class.isAssignableFrom(ForgeRegistries.BLOCKS.getValue(new ResourceLocation(origin, "blockhut" + hut)).getClass()))
```

Comment:
IAnchorBlock is definitely not a way for me to distinguish huts and the others
@OrionDevelopment meh, reworking this to something "normal" does not make much sense - the mod is one gigantic tape fix and every time you try to rewrite something you end up rewriting half of the mod, I'm completely fine with previous minor change

Reply:
I'm willing to revert this. My reasoning was that every hut block implements IAnchorBlock. I do realize that decoration controllers do so as well, but figured that they would be placed elsewhere by StructureLoadingUtils because of the path used .